runtime/interpreter/value.go

```diff
@@ -15009,16 +15163,27 @@ func (*EphemeralReferenceValue) DeepRemove(_ *Interpreter) {
 //
 type AddressValue common.Address
 
-func NewAddressValue(a common.Address) AddressValue {
-	return NewAddressValueFromBytes(a[:])
-}
-
-func NewAddressValueFromBytes(b []byte) AddressValue {
+func NewUnmeteredAddressValue(b []byte) AddressValue {
 	result := AddressValue{}
 	copy(result[common.AddressLength-len(b):], b)
 	return result
 }
 
+func NewAddressValue(memoryGauge common.MemoryGauge, a common.Address) AddressValue {
+	return NewAddressValueFromBytes(memoryGauge, a[:])
+}
+
+func NewAddressValueFromBytes(memoryGauge common.MemoryGauge, b []byte) AddressValue {
```

Comment:
The last parameter should be lazily evaluated, e.g. by passing it as a function, so that the metering happens before the actual allocation

Reply:
I should do this for all of the variable types, right?
- TypeValue
- CharacterValue
- StorageReferenceValue
- PathValue
- AddressValue

Reply:
Done.